app/models/preserved_copy.rb

```diff
@@ -30,4 +30,25 @@ class PreservedCopy < ApplicationRecord
   validates :size, numericality: { only_integer: true, greater_than: 0 }, allow_nil: true
   validates :status, inclusion: { in: statuses.keys }
   validates :version, presence: true
+
+  def update_audit_timestamps(moab_validated, version_audited)
+    t = Time.current
+    self.last_moab_validation = t if moab_validated
+    self.last_version_audit = t if version_audited
+  end
+
+  # moab_validated must not be nil. boolean indicating whether validation has been run (regardless of result).
+  # new_version is expected to be numeric
+  # new_size is expected to be numeric if provided (nil is allowed)
```

Comment:
super picky / not required:  I'm used to the @param notation in comments.

Reply:
ah yeah, good point, filed #490 for a bit of follow on work in this model